frame/glutton/src/lib.rs

```diff
@@ -105,6 +105,37 @@ pub mod pallet {
 	#[pallet::storage]
 	pub(crate) type TrashDataCount<T: Config> = StorageValue<_, u32, ValueQuery>;
 
+	#[pallet::genesis_config]
+	pub struct GenesisConfig {
+		pub compute: Perbill,
+		pub storage: Perbill,
+		pub trash_data_count: u32,
+	}
+
+	impl Default for GenesisConfig {
+		fn default() -> Self {
+			Self {
+				compute: Default::default(),
+				storage: Default::default(),
+				trash_data_count: Default::default(),
+			}
+		}
+	}
+
+	#[pallet::genesis_build]
+	impl<T: Config> GenesisBuild<T> for GenesisConfig {
+		fn build(&self) {
+			<Compute<T>>::put(self.compute);
+			<Storage<T>>::put(self.storage);
+
+			if self.trash_data_count <= 65_000 {
```

Comment:
That value is random? And the user will be confused why there is no trash data?

Reply:
> And the user will be confused why there is no trash data?
The trash data can be initialised from one of its main calls

Reply:
Okay, but then it should panic if the value is bigger than 65_000.

Reply:
> And the user will be confused why there is no trash data?

Yeah there should be an `else` statement to handle this (or just ensure before that it's below the pallet's limit).

Reply:
@ggwpez also brought up a good idea, we should `hash` the data before we put it into the `TrashData`. This should ensure that the data isn't compressable that much.

Reply:
right, I overlooked that

Reply:
Yes it is using TwoX now, since that is faster and I assume we dont need the cryptographic property of Blake.